owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.kt

```diff
@@ -0,0 +1,252 @@
+ *   @author David González Verdugo
+ *   Copyright (C) 2019 ownCloud GmbH
+ *
+ *   Permission is hereby granted, free of charge, to any person obtaining a copy
+ *   of this software and associated documentation files (the "Software"), to deal
+ *   in the Software without restriction, including without limitation the rights
+ *   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *   copies of the Software, and to permit persons to whom the Software is
+ *   furnished to do so, subject to the following conditions:
+ *
+ *   The above copyright notice and this permission notice shall be included in
+ *   all copies or substantial portions of the Software.
+ *
+ *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+
+package com.owncloud.android.lib.resources.shares
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.http.HttpConstants
+import com.owncloud.android.lib.common.http.methods.nonwebdav.PostMethod
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.utils.Log_OC
+import okhttp3.FormBody
+import java.net.URL
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+/**
+ * Creates a new share.  This allows sharing with a user or group or as a link.
+ *
+ * @author masensio
+ * @author David A. Velasco
+ * @author David González Verdugo
+ */
+class CreateRemoteShareOperation
+/**
+ * Constructor
+ *
+ * @param remoteFilePath Full path of the file/folder being shared. Mandatory argument
+ * @param shareType      0 = user, 1 = group, 3 = Public link. Mandatory argument
+ * @param shareWith      User/group ID with who the file should be shared.  This is mandatory for shareType
+ * of 0 or 1
+ * @param publicUpload   If false (default) public cannot upload to a public shared folder.
+ * If true public can upload to a shared folder. Only available for public link shares
+ * @param password       Password to protect a public link share. Only available for public link shares
+ * @param permissions    1 - Read only Default for public shares
+ * 2 - Update
+ * 4 - Create
+ * 8 - Delete
+ * 16- Re-share
+ * 31- All above Default for private shares
+ * For user or group shares.
+ * To obtain combinations, add the desired values together.
+ * For instance, for Re-Share, delete, read, update, add 16+8+2+1 = 27.
+ */
+    (
+    private val mRemoteFilePath: String,
```

Comment:
why don't you remove "m" prefix?

Reply:
We have historically followed the AOSP Java Code Style which says:
![screenshot 2019-03-05 at 09 24 58](https://user-images.githubusercontent.com/15261363/53790999-9aceda00-3f28-11e9-856a-36ef19de8094.png)

But since **our Java code will be eventually replaced with Kotlin, I will get rid of `m` prefixes**.